utils/flink-components-testkit/src/main/scala/pl/touk/nussknacker/engine/flink/util/test/FlinkTestScenarioRunner.scala

```diff
@@ -121,18 +122,27 @@ class FlinkTestScenarioRunner(
     // It's copied from registrar.register only for handling compilation errors..
     // TODO: figure how to get compilation result on highest level - registrar.register?
     val compiler = new FlinkProcessCompilerWithTestComponents(testExtensionsHolder, modelData, componentUseCase)
-    val compileProcessData = compiler.compileProcess(
-      scenario,
-      ProcessVersion.empty,
-      ProductionServiceInvocationCollector,
-      getClass.getClassLoader
-    )
 
-    compileProcessData.compileProcess().map { _ =>
-      val registrar = FlinkProcessRegistrar(compiler, ExecutionConfigPreparer.unOptimizedChain(modelData))
-      registrar.register(env, scenario, ProcessVersion.empty, DeploymentData.empty, testRunId = None)
-      env.executeAndWaitForFinished(scenario.id)()
-      RunUnitResult(errors = Nil)
+    Using.resource(TestScenarioCollectorHandler.createHandler(componentUseCase)) { testScenarioCollectorHandler =>
+      val compileProcessData = compiler.compileProcess(
+        scenario,
+        ProcessVersion.empty,
+        testScenarioCollectorHandler.resultCollector,
+        getClass.getClassLoader
+      )
+
+      compileProcessData.compileProcess().map { _ =>
+        val registrar = FlinkProcessRegistrar(compiler, ExecutionConfigPreparer.unOptimizedChain(modelData))
+        registrar.register(
+          env,
+          scenario,
+          ProcessVersion.empty,
+          DeploymentData.empty,
+          testRunId = testScenarioCollectorHandler.testRunId()
+        )
+        env.executeAndWaitForFinished(scenario.id)()
+        RunUnitResult(errors = Nil)
```

Comment:
I don't get it. Why we don't use anywhere `testScenarioCollectorHandler.results`? Am I right that we do all these abstracts only to avoid calling the "production" code and we don't use these tests results? How errors (e.g. dividing by zero in expression) will be handled? Won't they be handled by handler and hidden instead of returning to users? Can you do a test for that?

Reply:
Ekhm.. In the register method you can find funny @mproch comment: 
`// TODO: move creation outside Registrar, together with refactoring SinkInvocationCollector...` :)  We can  try to refactor it ;)